flake8 has new complaint after update

```
$ make check
flake8 --version
3.5.0 (flake-mutable: 1.2.0, flake8-docstrings: 1.3.0, pydocstyle: 2.1.1, flake8-mypy: 17.8.0, flake8_builtins: 1.4.1, mccabe: 0.6.1, pycodestyle: 2.3.1, pyflakes: 1.6.0) CPython 3.6.3 on Linux
flake8 --exclude=".travis,.eggs,__init__.py" --ignore=E203,W391,D107,A001,A002,A003,A004
./iocage/lib/Datasets.py:393:24: T484 The type alias to Union is invalid in runtime context
make: *** [check] Error 1
```

Found in the build log of a unrelated branch: [Travis build 403851235](https://travis-ci.org/iocage/libiocage/builds/403851235)

--- a/iocage/lib/Datasets.py
+++ b/iocage/lib/Datasets.py
@@ -390,7 +390,7 @@ class FilteredDatasets(Datasets):
             The shared logger instance.
     """
 
-    _source_filters = OptionalSourceFilterType
+    _source_filters: OptionalSourceFilterType
     datasets: Datasets
 
     def __init__(
@@ -404,6 +404,8 @@ class FilteredDatasets(Datasets):
         self.logger = iocage.lib.helpers.init_logger(self, logger)
         self.zfs = iocage.lib.helpers.init_zfs(self, zfs)
         self.datasets = datasets
+
+        self._source_filters = None
         self.source_filters = source_filters
 
     @property
@@ -415,7 +417,7 @@ class FilteredDatasets(Datasets):
     def source_filters(self, value: OptionalSourceFilterType) -> None:
         """Set or disable source filters."""
         self.clear()
-        self._source_filters = value  # noqa: T484
+        self._source_filters = value
         self._clone_from_datasets()
 
     def _clone_from_datasets(self) -> None:

